File: week_2__arrays/readability/readability.c
#include <cs50.h>
#include <stdio.h>
#include <math.h>

int main(void)
{
    // Coleman-Liau index
    string text = get_string("Text: ");

    // Count letters, words, and sentences
    int letters = 0;
    int words = 1;
    int sentences = 0;

    for (int i = 0; text[i] != '\0'; i++)
    {
        if ((text[i] >= 'a' && text[i] <= 'z') || (text[i] >= 'A' && text[i] <= 'Z'))
        {
            letters++;
        }
        else if (text[i] == ' ')
        {
            words++;
        }
        else if (text[i] == '.' || text[i] == '!' || text[i] == '?')
        {
            sentences++;
        }
        else if (text[i] == ',' || text[i] == ';' || text[i] == ':' || text[i] == '-')
        {
            continue;
        }
    }

    // Coleman-Liau index

    float L = ((float)letters / words) * 100;
    float S = ((float)sentences / words) * 100;
    int index = round(0.0588 * L - 0.296 * S - 15.8);

    // Print grade
    if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (index > 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %i\n", index);
    }
}

File: week_6__python/pset6/sentimental-readability/readability.py
def main():

    user_input = get_user_input()
    letters = count_letters(user_input)
    words = count_words(user_input)
    sentences = count_sentences(user_input)
    cli = coleman_liau_index(letters, words, sentences)
    print_result(cli)
    

def get_user_input():
    user_input = input("Text: ")
    return user_input


def count_letters(user_input):
    letters = filter(str.isalpha, user_input)
    letters = len(list(letters))
    print(letters)
    return letters


def count_words(user_input):
    words = user_input.split(" ")
    words = len(words)
    print(words)
    return words


def count_sentences(user_input):
    sentences = 0
    for i in range(len(user_input)):
        if user_input[i] == '.' or user_input[i] == '!' or user_input[i] == '?':
            sentences += 1
        elif user_input[i] == ',' or user_input[i] == ';' or user_input[i] == ':':
            continue
    print(sentences)
    return sentences


def coleman_liau_index(letters, words, sentences):
    L = letters * 100 / words
    S = sentences * 100 / words
    index = round(0.0588 * L - 0.296 * S - 15.8)
    return index


def print_result(cli):
    if cli < 1:
        print("Before Grade 1")
    elif cli >= 16:
        print("Grade 16+")
    else:
        print(f"Grade {cli}")


main()

